src/CHIPBindings.cc

```diff
@@ -224,7 +224,9 @@ hipError_t hipGraphReleaseUserObject(hipGraph_t graph, hipUserObject_t object,
   UNIMPLEMENTED(hipErrorNotSupported);
 }
 
-hipError_t hipInit(unsigned int flags) { return hipSuccess; };
+hipError_t hipInit(unsigned int flags) {
+  return flags ? hipErrorInvalidValue : hipSuccess;
```

Comment:
Following up on this, it seems we should use an argument (defaulting to false) to defer a full initialization (or defaulting to true to enforce a full initialization) to allow hipInit to still create the minimum state enabling error management support. Else calling `hipGetLastError` after `hipInit(1)` would return `hipSuccess`.
I can open an issue if this is the case.
Tagging @pvelesko here.

Reply:
Agreed, the `RETURN` macro should be used. I decided to do full runtime initialization at hipInit() call. It seems to be the right thing to do for getting the right answer in case the initialization fails (e.g. no valid platform available).